core/src/ldml/ldml_transforms.cpp

```diff
@@ -417,13 +416,16 @@ transform_entry::transform_entry(
     const kmx::kmx_plus &kplus,
     bool &valid)
     : fFrom(from), fTo(to), fFromPattern(nullptr), fMapFromStrId(mapFrom), fMapToStrId(mapTo) {
-  valid = false;
+  if (!valid)
+    return; // exit early
```

Comment:
This is similar to the status pattern from ICU. It means more work on the caller side but I guess I am okay with it!

Reply:
it makes sure the c'tor can return failure.  Could be a factory model, but the caller is using `emplace_back()` so the idea is a stack based construction anyway. on failure we're going to bail out of the entire load.